contracts/abstract/PrizeSplit.sol

```diff
@@ -0,0 +1,176 @@
+  }
+
+  /**
+    * @notice Update a prize split config.
+    * @dev Update a prize split config by passing a MultipleWinnersPrizeSplit struct and index position.
+    * @param prizeStrategySplit MultipleWinnersPrizeSplit config struct.
+    * @param prizeSplitIndex The target index to update.
+  */
+  function setPrizeSplit(MultipleWinnersPrizeSplit memory prizeStrategySplit, uint8 prizeSplitIndex) external onlyOwner {
+    require(prizeSplitIndex <= _prizeSplits.length.sub(1), "MultipleWinners/nonexistent-prizesplit");
+    require(prizeStrategySplit.target != address(0), "MultipleWinners/invalid-prizesplit-target");
+    
+    // Update the prize split config.
+    _prizeSplits[prizeSplitIndex] = prizeStrategySplit;
+
+    // Require the current prize split percentage does not exceed 100%.
+    uint256 totalPercentage = _totalPrizeSplitPercentageAmount();
+    require(totalPercentage <= 1000, "MultipleWinners/invalid-prizesplit-percentage-total");
+
+    // Emit the updated prize split configuration.
+    emit PrizeSplitSet(prizeStrategySplit.target, prizeStrategySplit.percentage, prizeStrategySplit.token, prizeSplitIndex);
+  }
+
+
+  /**
+    * @notice List of active prize splits
+    * @dev List of active prize splits set by the prize strategy owner
+    * @return _prizeSplits Array of MultipleWinnersPrizeSplit structs
+  */
+  function prizeSplits() external view returns (MultipleWinnersPrizeSplit[] memory) {
+    return _prizeSplits;
+  }
+
+  /**
+    * @notice Read a prize split config by index
+    * @dev Read a prize split MultipleWinnersPrizeSplit config by passing a _prizeSplits index
+    * @param prizeSplitIndex The target index to read
+    * @return MultipleWinnersPrizeSplit A single prize split config
+  */
+  function prizeSplit(uint256 prizeSplitIndex) external view returns (MultipleWinnersPrizeSplit memory) {
+    return _prizeSplits[prizeSplitIndex];
+  }
+
+  /**
+  * @dev Calculate the PrizeSplit percentage
+  * @param amount The prize amount
+  * @param percentage The prize split percentage amount
+  */
+  function _getPrizeSplitAmount(uint256 amount, uint16 percentage) internal pure returns (uint256) {
+    // Total prize split amount calculated from current award and target percentage.
+      return (amount * percentage).div(1000);
+  }
+
+  /**
+  * @notice Require the prize split percentage amount does not exceed 100%.
+  * @dev Require the prize split percentage amount does not exceed 100% with the current prize splits.
+  */
+  function _totalPrizeSplitPercentageAmount() internal view returns (uint256) {
+    uint256 _tempTotalPercentage;
+    for (uint8 index = 0; index < _prizeSplits.length; index++) {
+      MultipleWinnersPrizeSplit memory split = _prizeSplits[index];
+      _tempTotalPercentage = _tempTotalPercentage.add(split.percentage);
+    }
+    return _tempTotalPercentage;
+  }
+
+  function _distributePrizeSplits(uint256 prize) internal returns (uint256) {
```

Comment:
needs Natspec

Reply:
Will add for internal functions. Generally I avoid adding to internal function as it states in the documentation for natspec that internal functions are ignored for parsing.

> The Solidity compiler only interprets tags if they are external or public. You are welcome to use similar comments for your internal and private functions, but those will not be parsed.

https://docs.soliditylang.org/en/v0.5.10/natspec-format.html